### DISCUSSION TITLE
rke2-uninstall.ps1 fails with CTR.exe error

### DISCUSSION THREAD

**Discussion by Cojacfar** [rancher/rke2#4129] [Q&A]
**Environmental Info:**
RKE2 Version: 
```
rke2 -v
rke2.exe version v1.24.9+rke2r2 (2f4571a879954e1ea8d4560023eaf57c567df737)
go version go1.18.7b
```

Node(s) CPU architecture, OS, and Version: 
```
OS Name:                   Microsoft Windows Server 2019 Standard
OS Version:                10.0.17763 N/A Build 17763
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Member Server
OS Build Type:             Multiprocessor Free
System Manufacturer:       VMware, Inc.
System Model:              VMware Virtual Platform
System Type:               x64-based PC
Windows Directory:         C:\windows
System Directory:          C:\windows\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Total Physical Memory:     499,712 MB
Available Physical Memory: 471,400 MB
Virtual Memory: Max Size:  507,904 MB
Virtual Memory: Available: 473,800 MB
Virtual Memory: In Use:    34,104 MB
Network Card(s):           3 NIC(s) Installed.
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
```

Cluster Configuration:
This cluster is created from Rancher Manager 2.7.0, Windows nodes joined by default install.ps1 provided. 

**Describe the bug:**
Running rke2-uninstall.ps1 generates this error:
```
Beginning the uninstall process
Program 'ctr.exe' failed to run: The specified executable is not a valid application for this OS platform.At
C:\usr\local\bin\rke2-uninstall.ps1:274 char:9
+     if (ctr) {
+         ~~~.
At C:\usr\local\bin\rke2-uninstall.ps1:274 char:9
+     if (ctr) {
+         ~~~
    + CategoryInfo          : ResourceUnavailable: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : NativeCommandFailed
```

**Steps To Reproduce:**
- Install RKE2 on Windows Server with install.ps1 script downloaded from Rancher Manager generated join command. 
- Run rke2-uninstall.ps1 script

**Expected behavior:**
Expected the RKE2-Uninstall.ps1 to be able to clean up the node. 

Attempting to use the installed ctr.exe executable manually provides the same error - it seems this rke2 version is downloading a non-Windows version of CTR?

**Comment by brandond** [comment]
I'm not aware of any issues with the binary we're shipping. It is validated as part of the process of QAing every release. Perhaps the installation on your node was interrupted and the binary corrupted?

**Comment by Cojacfar** [comment]
I see this behavior across 12 different Windows nodes at the moment, so I don't think it's any kind of freak error in the download.

I have the ctr.exe on my local machine, if there's a way to get to you. I'm not sure what to look at in it to identify what's wrong. It gives the same error attempting to run it on my Windows 10 machine as on the 2019 Server machines.

**Comment by sadhugit** [comment]
I am also seeing similar behaviour while registering the windows node using Rancher 2.6.10 in one of the customer environment
